### PR TITLE
2022-end-of-year-update: Add post

### PR DESCRIPTION
**Pull request by Dennisbonke**
I'm planning on getting this merged on the 30th of december 2022 so it's still on time. Initial reviews would be appreciated.

### CODE REVIEW

File: _posts/2022-12-30-end-of-year-update.markdown [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+## What do we want to achieve in the next year?
+
+**Finish porting xbps.** Unfortunately, we didn't get exactly where we wanted to be this year on this point, even though the netstack got some nice updates and is a lot more stable now. We aim to have `xbps` in a state where an end user can just update the package repository and install packages by the end of next year, on supported network hardware.
+
+**Upstream port patches.** We made a good initial effort this year to start upstreaming patches, and we managed to [land our targets](https://lists.gnu.org/archive/html/config-patches/2022-09/msg00001.html) in the config.sub repo, used by autotools. In the coming year, we'd like to start upstreaming our targets to the respective toolchain packages (binutils, gcc, llvm, libtool), and if successful, start upstreaming patches to other packages.
```
Comment (64) [inline _posts/2022-12-30-end-of-year-update.markdown]: > to the respective toolchain packages (binutils, gcc, llvm, libtool)

\+ rust 😄

**Review by 64 — APPROVED** [review]

File: _posts/2022-12-30-end-of-year-update.markdown (hunk 1 above)
Comment (Dennisbonke) [reply _posts/2022-12-30-end-of-year-update.markdown]: I'll add it in.

File: _posts/2022-12-31-end-of-year-update.markdown [hunk 2]
```diff
@@ -49,26 +49,26 @@ When Qt6 came around, they switched to CMake, which is a lot nicer for cross-com
...
 
 ![gtk3-demo](/assets/2022-end-of-year-update/gtk3-demo.png)
 
 ## Other projects updates
 
 ### mlibc on Linux ([repo](https://github.com/no92/linux-mlibc))
-While working towards running a proper browser on Managarm, we came to the realization that we can't rule out whether the kernel / servers are wrong (A.K.A. Managarm itself), or if mlibc had a bug. To eliminate the first possibility and granting easy access to lots of nice debug tools, we started the [mlibc LFS](https://github.com/no92/linux-mlibc) project, named after Linux From Scratch. It's essentially Linux From Scratch, but using mlibc instead of glibc.
+While working towards running a proper browser on Managarm, we came to the realization that we can not rule out whether the kernel / servers are wrong (A.K.A. Managarm itself), or if mlibc had a bug. To eliminate the first possibility and granting easy access to lots of nice debug tools, we started the [mlibc LFS](https://github.com/no92/linux-mlibc) project, named after Linux From Scratch. It is essentially Linux From Scratch, but using mlibc instead of glibc.
```
Comment (64) [inline _posts/2022-12-31-end-of-year-update.markdown]: 'can not' is a bit weird

I would reword the first sentence: 'we struggled to determine whether mlibc was at fault, or the kernel / servers' (something like that)

Comment (64) [reply _posts/2022-12-31-end-of-year-update.markdown]: Also 'but using' -> 'but uses'

File: _posts/2022-12-31-end-of-year-update.markdown [hunk 3]
```diff
@@ -49,26 +49,26 @@ When Qt6 came around, they switched to CMake, which is a lot nicer for cross-com
...
 
 ## What do we want to achieve in the next year?
 
-**Finish porting xbps.** Unfortunately, we didn't get exactly where we wanted to be this year on this point, even though the netstack got some nice updates and is a lot more stable now. We aim to have `xbps` in a state where an end user can just update the package repository and install packages by the end of next year, on supported network hardware.
+**Finish porting xbps.** Unfortunately, we did not get exactly where we wanted to be this year on this point, even though the netstack is a lot more stable now. We aim to have `xbps` in a state where an end user can just update the package repository and install packages by the end of next year, on supported network hardware.
```
Comment (64) [inline _posts/2022-12-31-end-of-year-update.markdown]: First sentence reads a bit clunky, could just be 'Unfortunately, despite improvements to the network stack, we did not finish porting `xbps` this year'.

File: _posts/2022-12-31-end-of-year-update.markdown [hunk 4]
```diff
@@ -49,26 +49,26 @@ When Qt6 came around, they switched to CMake, which is a lot nicer for cross-com
...
+**Port a web browser and work towards self hosting.** While we made great improvements in this area, including the addition of the `links` browser mentioned in the August update, we aim to have one of the common browsers (Chrome, Firefox, WebKitGTK) running by the end of 2023. We also would like to get more common development programs in, like `git`. While we also had this goal at the end of 2019, it turned out that the given timeframe too ambitious. However, due to the progress achieved in the past three years, like the porting of the X stack and GTK 3, and the addition of someone to the team who likes to work on porting software, we feel that we can get this goal done in 2023.
```
Comment (64) [inline _posts/2022-12-31-end-of-year-update.markdown]: 'too ambitious' -> 'was too ambitious'

Comment (64) [reply _posts/2022-12-31-end-of-year-update.markdown]: 'the porting of' -> 'porting the'

File: _posts/2022-12-31-end-of-year-update.markdown [hunk 5]
```diff
@@ -49,26 +49,26 @@ When Qt6 came around, they switched to CMake, which is a lot nicer for cross-com
...
 
-**Add more drivers.** Specifically graphics and networking related. Together with a few [LAI](https://github.com/managarm/lai) quirks and USB related quirks, this is the biggest issue for running on real hardware, a lack of drivers. Recently, work has been put into writing more network drivers, and some members are looking into adding at least modesetting drivers for Intel iGPUs. This is certainly an area we'd like help in, even if it is just testing on real hardware for quirks.
+**Add more drivers.** Specifically graphics and networking related. Together with a few [LAI](https://github.com/managarm/lai) quirks and USB related quirks, a lack of drivers is the biggest issue for running on real hardware. Recently, work has been put into writing more network drivers, and some contributors are looking into adding at least modesetting drivers for Intel iGPUs. This is certainly an area we would benefit from additional contributions, even if it is just testing on real hardware for quirks.
```
Comment (64) [inline _posts/2022-12-31-end-of-year-update.markdown]: don't repeat the word 'quirks'

Comment (64) [reply _posts/2022-12-31-end-of-year-update.markdown]: 'at least modesetting' just say 'modesetting'

Comment (64) [reply _posts/2022-12-31-end-of-year-update.markdown]: 'an area we' -> 'an area where we'

Comment (64) [reply _posts/2022-12-31-end-of-year-update.markdown]: 'even if it is just testing on real hardware for quirks' -> 'even just testing on real hardware'

Comment (Dennisbonke) [reply _posts/2022-12-31-end-of-year-update.markdown]: Applied, thanks!

File: _posts/2022-12-31-end-of-year-update.markdown (hunk 4 above)
Comment (Dennisbonke) [reply _posts/2022-12-31-end-of-year-update.markdown]: Applied, thanks!

File: _posts/2022-12-31-end-of-year-update.markdown (hunk 3 above)
Comment (Dennisbonke) [reply _posts/2022-12-31-end-of-year-update.markdown]: Applied, thanks!

File: _posts/2022-12-31-end-of-year-update.markdown (hunk 2 above)
Comment (Dennisbonke) [reply _posts/2022-12-31-end-of-year-update.markdown]: Applied, thanks!

File: _posts/2022-12-31-end-of-year-update.markdown [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+![gtk3-demo](/assets/2022-end-of-year-update/gtk3-demo.png)
+
+## Other projects updates
+
+### mlibc on Linux ([repo](https://github.com/no92/linux-mlibc))
+While working towards running a proper browser on Managarm, we struggled to determine wether mlibc was at fault, or wether the Managarm kernel or its servers were wrong. To eliminate the first possibility and granting easy access to lots of nice debug tools, we started the [mlibc LFS](https://github.com/no92/linux-mlibc) project, named after Linux From Scratch. It is essentially Linux From Scratch, but uses mlibc instead of glibc.
```
Comment (64) [inline _posts/2022-12-31-end-of-year-update.markdown]: wether -> whether

File: _posts/2022-12-31-end-of-year-update.markdown [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+We quickly got a small userspace running in a chroot, but it eventually grew out to something that we can boot up in a VM, a neccesity when testing out graphical programs. The major things that came from this still ongoing project is the fixes and additions to the mlibc Linux ABI, general usability under Linux, and an easier way to debug applications in the future. While we did get to a point where we could run WebKitGTK reasonably, this has since regressed on a nasty deadlock in pthreads, which we're working on fixing.
```
Comment (64) [inline _posts/2022-12-31-end-of-year-update.markdown]: is the fixes -> are the fixes

File: _posts/2022-12-31-end-of-year-update.markdown [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+**Upstream port patches.** We made a good initial effort this year to start upstreaming patches, and we managed to [land our targets](https://lists.gnu.org/archive/html/config-patches/2022-09/msg00001.html) in the GNU config repo, used by autotools. In the coming year, we would like to start upstreaming our targets to the respective toolchain packages (binutils, GCC, LLVM, libtool and rustc), and if successful, start upstreaming patches to other packages.
+
+**Port a web browser and work towards self hosting.** While we made great improvements in this area, including the addition of the `links` browser mentioned in the August update, we aim to have one of the common browsers (Chrome, Firefox, WebKitGTK) running by the end of 2023. We also would like to get more common development programs in, like `git`. While we also had this goal at the end of 2019, it turned out that the given timeframe was too ambitious. However, due to the progress achieved in the past three years, like porting the X stack and GTK 3, and the addition of someone to the team who likes to work on porting software, we feel that we can get this goal done in 2023.
```
Comment (64) [inline _posts/2022-12-31-end-of-year-update.markdown]: get this goal done -> achieve this goal

File: _posts/2022-12-31-end-of-year-update.markdown [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+**Add more drivers.** Specifically graphics and networking related. Together with a few [LAI](https://github.com/managarm/lai) and USB related quirks, a lack of drivers is the biggest issue for running on real hardware. Recently, work has been put into writing more network drivers, and some contributors are looking into adding modesetting drivers for Intel iGPUs. This is certainly an area where we would benefit from additional contributions, even just testing on real hardware.
```
Comment (64) [inline _posts/2022-12-31-end-of-year-update.markdown]: sorry, changed my mind
I think 'even just testing on real hardware' makes more sense as 'even just further testing on real hardware'

File: _posts/2022-12-31-end-of-year-update.markdown [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+## Help wanted!
+
+We are always looking for new contributors. If you want to get involved in the project, you can search our [GitHub issues](https://github.com/managarm/managarm/issues) for interesting tasks. Aside from contributions to the kernel and/or POSIX layer, we would be particularly interested in drivers for additional file systems (other than ext2), since Managarm currently lacks good drivers in these areas, and better debugging tools, like `ptrace` and related API's. As always, just testing on real hardware to find quirks or test drivers is also an excellent way to help us!
```
Comment (64) [inline _posts/2022-12-31-end-of-year-update.markdown]: API's -> APIs
